docs/working-with-pytrees.md

```diff
@@ -272,6 +272,51 @@ jax.tree.leaves([
 
 Notice that the `name` field now appears as a leaf, because all tuple elements are children. This is what happens when you don't have to register the class the hard way.
 
+Unlike `NamedTuple` subclasses, classes decorated with `@dataclass` are not automatically pytrees. However, they can be registered as pytrees using the {func}`jax.tree_util.register_dataclass` decorator:
+
+```{code-cell}
+from dataclasses import dataclass
+import functools
+
+@functools.partial(jax.tree_util.register_dataclass,
+                   data_fields=['a', 'b', 'c'],
+                   meta_fields=['name'])
+@dataclass
+class MyDataclassContainer(object):
+  name: str
+  a: Any
+  b: Any
+  c: Any
+
+# MyDataclassContainer is now a pytree node.
+jax.tree.leaves([
+  MyDataclassContainer('apple', 5.3, 1.2, jnp.zeros([4])),
+  MyDataclassContainer('banana', np.array([3, 4]), -1., 0.)
+])
+```
+
+Notice that the `name` field does not appear as a leaf. This is because we included it in the `meta_fields` argument to {func}`jax.tree_util.register_dataclass`, indicating that it should be treated as metadata/auxiliary data, just like `aux_data` in `RegisteredSpecial` above. Now instances of `MyDataclassContainer` can be passed into JIT-ed functions, and `name` will be treated as static (see [Just-in-time compilation](https://jax.readthedocs.io/en/latest/jit-compilation.html#marking-arguments-as-static) for more information on static args):
```

Comment:
Rather than the URL, we could use a cross reference: writing ``` (see {ref}`jit-compilation` for more information...) ``` should generate a validated link here.

Reply:
Done (with a tag to the "Marking arguments as static" section of the JIT tutorial).